src/lib/DefaultRuleManager.ts

```diff
@@ -236,46 +236,58 @@ export class DefaultRuleManager implements RuleManager {
 		const pm = new PathMatcher([...engineTargets, ...negativePatterns]);
 		for (const target of positivePatterns) {
 			// Used to detect if the target resulted in a match
-			const startLength: number = ruleTargets.length;
+			const ruleTargetsInitialLength: number = ruleTargets.length;
+			// Positive patterns might use method-level targeting. We only want to do path evaluation against the part
+			// that's actually a path.
+			const targetPortions = target.split('#');
```

Comment:
Needed to make sure that we're always globbing the actual file path, rather than the file path plus any method targets.

Reply:
Is there any risk of another engine type invoking with `#` in `--target` value and us not rejecting it?